nilearn/datasets/tests/test_func.py

```diff
@@ -756,15 +756,16 @@ def test_fetch_development_fmri_exception():
 
 # datasets tests originally belonging to nistats follow
 
-currdir = os.path.dirname(os.path.abspath(__file__))
+currdir = os.path.dirname(Path.resolve(__file__))
 datadir = os.path.join(currdir, "data")
 
 
 def test_fetch_bids_langloc_dataset(tmp_path):
     data_dir = str(tmp_path / "bids_langloc_example")
-    os.mkdir(data_dir)
+    main_folder = tmp_path / "bids_langloc_example / "bids_langloc_dataset"
```

Comment:
```suggestion
    main_folder = tmp_path / "bids_langloc_example" / "bids_langloc_dataset"
```

oops my bad forgot a `"`